recipes/googleapis/all/conanfile.py

```diff
@@ -11,18 +18,126 @@ class GoogleAPIS(ConanFile):
     homepage = "https://github.com/googleapis/googleapis"
     topics = "google", "protos", "api"
     settings = "os", "arch", "compiler", "build_type"
-    short_paths = True
+    generators = "cmake", "cmake_find_package"
+    options = {
+        "shared": [True, False], 
+        "fPIC": [True, False]
+        }
+    default_options = {
+        "shared": False, 
+        "fPIC": True
+        }
+    exports = "helpers.py"
+
+    def layout(self):
+        cmake_layout(self)
+
+    def export_sources(self):
+        self.copy("CMakeLists.txt")
 
     def source(self):
         get(self, **self.conan_data["sources"][str(self.version)], destination=self.source_folder, strip_root=True)
 
-    def package_id(self):
-        self.info.header_only()
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 11)
+        if self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) <= "5":
+            raise ConanInvalidConfiguration("Build with GCC 5 fails")
```

Comment:
If I have done it right locally, it fails using `conanio/gcc5:1.50.0`, but not using `conanio/gcc5_ubuntu16.04:1.50.0` 🤷